docs/src/main/sphinx/admin/properties-writer-scaling.rst

```diff
@@ -33,3 +33,11 @@ another writer is eligible to be added. Each writer task may have multiple
 writers, controlled by ``task.writer-count``, thus this value is effectively
 divided by the number of writers per task. This can be specified on a
 per-query basis using the ``writer_min_size`` session property.
+
+``use-preferred-write-partitioning``
+^^^^^^^^^^^^^^^^^^^
+* **Type:** ``boolean``
+* **Default value:** ``false``
+
+Enables preferred write partitioning. This option will force every partition
```

Comment:
this suggest that all partitions will be written by single worker, which is not the case.
I suggest to extract this commit as separate PR and tag @mosabua

Reply:
that is a needed feature when writing small dataset!